tensorboard/plugins/debugger/session_debug_test.py

```diff
@@ -0,0 +1,339 @@
+    for _ in range(max_tries):
+      try:
+        with tf.Session() as sess:
+          a_init_val = np.array([42.0])
+          a_init = tf.constant(a_init_val, shape=[1], name="a_init")
+          a = tf.Variable(a_init, name="a")
+
+          run_options = tf.RunOptions(output_partition_graphs=True)
+          tf_debug.watch_graph(run_options,
+                               sess.graph,
+                               debug_ops=["DebugNumericSummary"],
+                               debug_urls=[self._debug_url])
+
+          sess.run(a.initializer, options=run_options)
+          poll_success = True
+          break
+      except tf.errors.FailedPreconditionError:
+        time.sleep(poll_interval_seconds)
+
+    return poll_success
+
+  def _compute_health_pill(self, x):
+    x_clean = x[np.where(
+        np.logical_and(
+            np.logical_not(np.isnan(x)), np.logical_not(np.isinf(x))))]
+    if np.size(x_clean):
+      x_min = np.min(x_clean)
+      x_max = np.max(x_clean)
+      x_mean = np.mean(x_clean)
+      x_var = np.var(x_clean)
+    else:
+      x_min = np.inf
+      x_max = -np.inf
+      x_mean = np.nan
+      x_var = np.nan
+
+    return np.array([
+        1.0,  # Assume is initialized.
+        np.size(x),
+        np.sum(np.isnan(x)),
+        np.sum(x == -np.inf),
+        np.sum(np.logical_and(x < 0.0, x != -np.inf)),
+        np.sum(x == 0.0),
+        np.sum(np.logical_and(x > 0.0, x != np.inf)),
+        np.sum(x == np.inf),
+        x_min,
+        x_max,
+        x_mean,
+        x_var,
+        float(tf.as_dtype(x.dtype).as_datatype_enum),
+        float(len(x.shape)),
+    ] + list(x.shape))
+
+  def _check_health_pills_in_events_file(self,
+                                         events_file_path,
+                                         debug_key_to_tensors):
+    reader = tf.python_io.tf_record_iterator(events_file_path)
+    event_read = tf.Event()
+
+    # The first event in the file should contain the events version, which is
+    # important because without it, TensorBoard may purge health pill events.
+    event_read.ParseFromString(next(reader))
+    self.assertEqual("brain.Event:2", event_read.file_version)
+
+    num_events_read = 0
+    health_pills = {}
+    while True:
+      try:
+        event_read.ParseFromString(next(reader))
+        num_events_read += 1
+        values = event_read.summary.value
+        if values:
+          plugin = constants.DEBUGGER_PLUGIN_NAME
+          if (values[0].metadata.plugin_data and
+              values[0].metadata.plugin_data[0].plugin_name == plugin):
+            debug_key = values[0].node_name
+            if debug_key not in health_pills:
+              health_pills[debug_key] = [
+                  tf_debug.load_tensor_from_event(event_read)
+              ]
+            else:
+              health_pills[debug_key].append(
+                  tf_debug.load_tensor_from_event(event_read))
+      except StopIteration:
+        break
```

Comment:
You can simplify this by instead using `next_event = next(reader, None); if next_event is None: break; event_read.ParseFromString(next_event)` above. This limits the scope of the `try` block—e.g., you will not silently hide errors in `tf_debug.load_tensor_from_event`.

(I'm assuming from context that `None` is not a valid event. Otherwise: `sentinel = object(); next_event = next(reader, sentinel); if next_even is sentinel: break;`.)